File: redsheet-portal/src/app/database-models/lite-event-planning-action.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { LiteMeetingManagement, vLiteEventPlanningActionRecord } from './'
export class LiteEventPlanningAction {
    constructor(liteEventPlanningAction?: LiteEventPlanningAction | vLiteEventPlanningActionRecord) {
        let properties = ["liteEventPlanningActionBy", "liteEventPlanningActionDetail", "liteEventPlanningActionId", "liteEventPlanningActionOn", "liteMeetingManagementId",];
        for (let property of properties)
            if (liteEventPlanningAction && liteEventPlanningAction[property])
                this[property] = liteEventPlanningAction[property];
    }

    @required()
    @maxLength(100)
    liteEventPlanningActionBy: string = undefined;

    @required()
    @maxLength(200)
    liteEventPlanningActionDetail: string = undefined;

    liteEventPlanningActionId: number = 0;

    @required()
    liteEventPlanningActionOn: Date = undefined;

    @range(0, 2147483647)
    liteMeetingManagementId: number = undefined;
    liteMeetingManagement: LiteMeetingManagement;


}

File: RedSheet.DbEntities/Models/LiteEventPlanningAction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("LiteEventPlanningActions", Schema = "dbo")]
    [Serializable]
    public partial class LiteEventPlanningAction
    {

        [Required]
        [MaxLength(100)]
        public string LiteEventPlanningActionBy { get; set; }

        [Required]
        [MaxLength(200)]
        public string LiteEventPlanningActionDetail { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int LiteEventPlanningActionId { get; set; }

        [Required]
        public System.DateTime LiteEventPlanningActionOn { get; set; }

        [RelationshipTableAttribue("LiteMeetingManagements", "dbo")]
        //Relationships
        public int LiteMeetingManagementId { get; set; }

        [ForeignKey("LiteMeetingManagementId")]
        public LiteMeetingManagement LiteMeetingManagement { get; set; }

        public LiteEventPlanningAction()
        {
        }

    }
}

